src/utils/string/StringUtils.js

```diff
@@ -398,6 +397,10 @@ const onOpenLink = (url) => {
   });
 };
 
+const isValidUrl = (url) => {
+  return urlRegex.test(url);
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isValidUrl` function uses `urlRegex.test(url)`, which can lead to unexpected results due to the stateful nature of JavaScript RegExp objects when used with the `g` flag. If you use `test()` multiple times on the same global RegExp object, subsequent tests may start from where the last match ended, leading to inconsistent results. Consider removing the `g` flag from the `urlRegex` definition or use a different approach for testing the regex.
```diff
- const urlRegex = /(https?:\/\/\S+)+/g;
+ const urlRegex = /(https?:\/\/\S+)/;

- return urlRegex.test(url);
+ return url.match(urlRegex) !== null;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->